### PR TITLE
(GH-14) Remove dependency on JSON.NET

### LINKED ISSUE #14: Remove dependency on JSON.NET
We should get rid of the dependency to JSON.NET, so that we can target .NET Standard, which is not supported by Fody Costura. Could be replaced by `DataContractJsonSerializer` like in [Cake.Issues.EsLint](https://github.com/cake-contrib/Cake.Issues.EsLint/pull/7)

### PR DESCRIPTION
**Pull request by rinukkusu**
Remove dependency on JSON.NET

Fixes #14

### CODE REVIEW

**Comment by pascalberger** [comment]
@rinukkusu Thanks for your contribution 👍 

What are the changes in the testfiles and the reason of the change?

**Comment by rinukkusu** [comment]
I removed the UTF-8 BOM since the `DataContractJsonSerializer` had problems deserializing that.

**Comment by pascalberger** [comment]
@rinukkusu `docfx.json` is a file exactly as written by DocFx (the other ones are manually created). So the actual files will contain a BOM and could should be able to handle this.

File: src/Cake.Issues.DocFx/DocFxIssuesProvider.cs [hunk 1]
```diff
@@ -36,38 +38,48 @@ protected override IEnumerable<IIssue> InternalReadIssues(IssueCommentFormat for
                 docRootPath = docRootPath.MakeAbsolute(this.Settings.RepositoryRoot);
             }
 
+            IEnumerable<LogEntryDataContract> logFileEntries = null;
+
+            var logFileContent = Encoding.UTF8.GetString(this.IssueProviderSettings.LogFileContent)
```
Comment (pascalberger) [inline src/Cake.Issues.DocFx/DocFxIssuesProvider.cs]: You can use the [ToStringUsingEncoding extension method](https://cake-contrib.github.io/Cake.Issues.Website/api/Cake.Issues/ByteArrayExtensions/B203630E) with passing `true` for the `skipPreamble` argument.

Comment (rinukkusu) [reply src/Cake.Issues.DocFx/DocFxIssuesProvider.cs]: Oh, handy 😄

**Comment by pascalberger** [comment]
@rinukkusu Your changes have been merged. Thanks again for your contribution 👍

**Pull request merged**